backend/common/utils/spatial.py

```diff
@@ -0,0 +1,158 @@
+        """
+        width, height = image_size
+        new_dimension = min(width, height)
+        left = (width - new_dimension) / 2
+        upper = (height - new_dimension) / 2
+        right = (width + new_dimension) / 2
+        lower = (height + new_dimension) / 2
+        return (left, upper, right, lower)
+
+    def _prepare_image(self, content):
+        """
+        Prepare the image array for processing.
+
+        Args:
+            content (dict): The content dictionary containing the image array.
+
+        Returns:
+            np.ndarray: The prepared image array.
+        """
+        resolution = "fullres" if "fullres" in content["images"] else "hires"
+        image_array = content["images"][resolution]
+        image_array_uint8 = np.uint8(image_array * 255 if resolution == "hires" else image_array)
+        return image_array_uint8
+
+    def _process_and_flip_image(self, image_array_uint8):
+        """
+        Process and flip the image.
+
+        Args:
+            image_array_uint8 (np.ndarray): The image array.
+
+        Returns:
+            np.ndarray: The processed and flipped image array.
+        """
+        Image.MAX_IMAGE_PIXELS = None  # Disable the image size limit
+        try:
+            with Image.fromarray(image_array_uint8) as img:
+                # Convert image to RGB mode if it's not already
+                if img.mode != "RGB":
+                    img = img.convert("RGB")
+                cropped_img = img.crop(self._calculate_aspect_ratio_crop(img.size))  # Crop the image
+                cropped_img.save(io.BytesIO(), format="JPEG", quality=90)  # Save or manipulate as needed
+                # Flip the image vertically due to explorer client rendering images upside down
+                flipped_img = cropped_img.transpose(Image.FLIP_TOP_BOTTOM)
+                return np.array(flipped_img)
+        except Exception as e:
+            logger.error(f"Error processing image: {e}")
+            raise Exception(f"Error processing image: {e}") from e
+
+    def _generate_deep_zoom_assets(self, image_array, folder_name):
+        """
+        Generate deep zoom assets from the image array.
+
+        Args:
+            image_array (np.ndarray): The image array.
+            folder_name (str): The name of the folder to save the assets.
+        """
+        h, w, bands = image_array.shape
+        linear = image_array.reshape(w * h * bands)
+        image = pyvips.Image.new_from_memory(linear.data, w, h, bands, "uchar")
+        image.dzsave(folder_name + "spatial", suffix=".jpeg")
+
+    def _upload_assets(self, assets_folder):
+        """
+        Upload the deep zoom assets to the S3 bucket.
+
+        Args:
+            assets_folder (str): The folder containing the assets.
+        """
+        s3_uri = f"s3://{self.bucket_name}/{self.asset_directory}/{assets_folder}"
+        self.s3_provider.upload_directory(assets_folder, s3_uri)
+        shutil.rmtree(assets_folder)
```

Comment:
suggestion: rather than call this explicitly, use a [TemporaryDirectory](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) context manager. It will cleanup the files when exiting the context. I find to be cleaner.